src/app/cli/src/commands/delete_command.rs

```diff
@@ -54,16 +57,29 @@ impl Command for DeleteCommand {
 
         let dataset_refs: Vec<_> = if self.all {
             unimplemented!("Recursive deletion is not yet supported")
-        } else if self.recursive {
-            unimplemented!("Recursive deletion is not yet supported")
         } else {
-            filter_datasets_by_pattern(
+            let dataset_ids: Vec<_> = filter_datasets_by_pattern(
                 self.dataset_repo.as_ref(),
                 self.dataset_ref_patterns.clone(),
             )
-            .map_ok(|dataset_handle| dataset_handle.as_local_ref())
+            .map_ok(|dataset_handle| dataset_handle.as_local_ref().id().unwrap().clone())
```

Comment:
I think this whole block should return a `Vec<DatasetHandle>`:
- You already have handles after filtering by pattern, so by mapping them to IDs you lose information
- You later need handles (dataset aliases) to ask for user confirmation
- I assume you mapped everything to IDs only because `dependency_graph_service` returns IDs, but we could resolve the result of `get_all_downstream_dependencies` to handles instead.

Reply:
Looks like it's also tied to `get_all_downstream_dependencies` expecting `Vec<DatasetID>` as input, and we probably don't want to complicate the dependency graph API with `DatasetRef`s ....

I guess let's keep it as is unless we think of a better solution.